src/cpp/core/DDSRouter.cpp

```diff
@@ -254,9 +254,15 @@ void DDSRouter::init_participants_()
                                                            << " and type " << new_participant->type() << ".");
 
         // Add this participant to the database
-        participants_database_->add_participant_(
+        if (!participants_database_->add_participant_(
             new_participant->id(),
-            new_participant);
+            new_participant))
+        {
+            // Duplicated Ids
+            logError(DDSROUTER, "Participant Ids must be unique.");
```

Comment:
I'd throw the exception in `add_participant_()` as it is the function performing the actual check. What it is done now mixes the return codes (true/false in this case) with exceptions.

Reply:
Actually it does not mix them because the ParticipantDatabase does not throw any exception at any case.
I think this was designed like that so in the future it may be able to update a participant, but I also think it could be an exception.